### ISSUE TITLE
twitter.com

### ISSUE THREAD

**Issue by deletescape** [catppuccin/catppuccin#172]
**Are you available to help in the creation of this port? 😄 (optional)**
Maybe

**Additional context 📷 (optional)**
It probably shouldn't be too hard to make a catppuccin stylus theme for twitter for someone with a bit of experience.

**Comment by Spaxly** [comment]
Wouldn't the Dark Reader extension be able to this? The extension does have the catppuccin colorscheme built in for usage.

**Comment by andreasgrafen** [comment]
Technically, yes. But making a dedicated theme usually is more precise and thus looks better. I've been looking at this yesterday for a bit, and it seems possible to do – even tho Twitter has some really annoying way to colour things.

Maybe I'll set up something and count on the community to jump in and make it better. :D

**Comment by nonetrix** [comment]
> Dark Reader

since when? I only have normal dark theme

**Comment by xor** [comment]
+1 I'd love to use theme for Twitter.

**Comment by andreasgrafen** [comment]
Well, where getting somewhere. slowly but surely. Twitters markup is … annoying, to phrase it mildly. I have no idea how persistant the randomly generated class names are, that's what scares me the most rn.

![image](https://user-images.githubusercontent.com/35840154/189884722-195b9207-2cd0-4a70-bdfb-bb64f9b54766.png)

**Comment by nonetrix** [comment]
Seems it has already somewhat broken 
![Screenshot 2022-09-21 at 02-49-08 (4) Home _ Twitter](https://user-images.githubusercontent.com/45698918/191446496-5883b6c5-a6dc-4edc-b72a-a8e8d3785b9c.png)

**Comment by RoootTheFox** [comment]
I currently have a theme that looks pretty good
It's still not finished but it is definitely getting there!
![image](https://user-images.githubusercontent.com/78933889/208545027-8a7479ed-8b8d-4fd7-8674-ff25e399c8f9.png)

I can put the source code into a repository or share it here if wanted/needed.

**Comment by RoootTheFox** [comment]
And regarding the random CSS IDs - the best bet would be to rely on other element properties such as HTML properties - I am using this to theme some elements that I couldn't figure out how to theme otherwise.
 ```css
[style^="color: rgb(29, 155, 240)"] {
   color: rgb(var(--ct-color-accent)) !important;
}
[style^="background-color: rgb(29, 155, 240)"] {
   background-color: rgb(var(--ct-color-accent)) !important;
}
 ```

**Comment by nonetrix** [comment]
Looks great 
![image](https://user-images.githubusercontent.com/45698918/209068497-9d4104c7-e402-4e1c-ad40-e7076a49b6c7.png)
I like the blur you added to many of the UI elements perhaps you could do something like this to add blur behind images
![image](https://user-images.githubusercontent.com/45698918/209069236-d584cb3f-a32a-436d-a271-ce79f3b44b9f.png)

**Comment by RoootTheFox** [comment]
Thank you for the idea, I have updated the stylesheet to add background blur to images!

**Comment by nonetrix** [comment]
Also, changing the new notification color looks good
![image](https://user-images.githubusercontent.com/45698918/215292113-4ca8ca4a-360b-4e39-b724-6d56885c07ba.png)

**Comment by sgoudham** [comment]
Is there anyway we can make this userstyle official? It would be great if it was transferred into a repository that people can add to and eventually it can make its way into `ready for review`?

**Comment by sgoudham** [comment]
Yeah it would be good if we can move the current theme into a repo and people are easily able to PR and work on it there

**Comment by RoootTheFox** [comment]
I'd be down to make a repo and implementing the flavours shouldn't be difficult because I made everything based on variables already

**Comment by RoootTheFox** [comment]
Made a repo (https://github.com/RoootTheFox/ctp-twitter), currently doesn't have the flavours and previews yet but everything else is there.

**Comment by sgoudham** [comment]
Thanks for taking the lead on that! @RoootTheFox Hopefully we can get this into a state soon where it's good to move over!

As long as Elon doesn't keep breaking things 👀
